charms/istio-pilot/config.yaml

```diff
@@ -17,6 +17,11 @@ options:
     type: string
     default: istio-gateway
     description: Name to use as a default gateway
+  domain-name:
```

Comment:
I'm trying to understand when this is needed, and also I am very networking ignorant :)

Sometimes we've seen LBs give the ingress service an external "ip" that is really a hostname.  In those cases, this setting is not needed as we can get the hostname straight from the service (?).

But iiuc, in some cases this is not true and the LB will provide an external IP even though the user will reach it by some hostname.  And in those cases, we need a way to tell istio-pilot (the charm responsible for talking with the cert provider) this hostname because it not automatically discoverable?

Reply:
I thought that, but then realised that our charm could actually discover hostnames and IP addresses from the `istio-ingressgateway-workload` assuming that the DNS resolves the hostname to the LB Service address, or in all other cases we'll try with the IP address.

I will update this PR removing the configuration option and letting the charm automatically discover this address.